### ISSUE TITLE
"Implied" Test Types

### ISSUE THREAD

**Issue by samm82** [samm82/TestGen-Thesis#23]
IEEE 2017 provides definitions of many software qualities. Since a test type is "Testing that is focused on specific quality characteristics" (e.g., functional testing, usability testing, and performance testing) (IEEE, 2022, p. 15) (see #21), it seems reasonable to consider each quality as implying a test type specific to it. Should I be including entries based on these qualities in my glossary? I've found the following (I've gone through A-E in the glossary so far 😅):

- adaptability (p. 12)
- availability (p. 38)
- cohesion (p. 74)
- comfort (p. 75) e.g., screen brightness
- compliance (p. 82)
- conciseness (p. 88)
- concurrency (p. 88)
- conformance (p. 92); may be at the process level, not the software level (see #22)
- connectivity (p. 93)
- consistency (p. 94)
- correctability (p. 104); related to maintainability?
- correctness (p. 104); precedence for this as a type of testing in ([Pan, 1999](https://gitlab.cas.mcmaster.ca/crawfs1/TestGen-Thesis-Papers/-/blob/main/Pan1999.pdf?ref_type=heads))
- coupling (p. 107)
- criticality? (p. 110)
- efficiency (p. 152); "performance efficiency testing" mentioned in IEEE (2022, p. 9; 2017, pp. 58, 159, 253, 442, 508)
- error tolerance (p. 166)
- expandability (p. 173)
- extendability (p. 174)

And perhaps a quality the software _shouldn't_ have:

- complexity (p. 81)

**Comment by JacquesCarette** [comment]
> it seems reasonable to consider each quality as implying a test type specific to it.

Agreed!

> Should I be including entries based on these qualities in my glossary? 

If you mean 'rows in your table', this feels like overkill, somehow. And yet...  Keeping a table of qualities does seem like a good idea.

**Comment by smiths** [comment]
As I mentioned in #21, functional isn't a quality.  The other qualities that are listed are qualities.

I agree that test types and qualities are related.  It sounds like you are saying each quality implies one test type.  I think it is possible for more than one test type?  It seems like there could be test types that are associated to more than one quality.  For instance, a performance test is implicitly also a usability test, since usability is typically improved when performance improves.

I don't think we want to go down the road of focusing too much on qualities.  @JacquesCarette and I did an exercise where we tried to define software qualities and it turned into a big headache.  We don't want to bring in the headache of defining quality and defining testing.  :smile:  

I agree that it is a good idea to keep a table of qualities.  We just don't want to go too far down the road defining and categorizing every little detail.  :smile:
